engine/storage/src/main/java/org/apache/cloudstack/storage/allocator/AbstractStoragePoolAllocator.java

```diff
@@ -82,6 +86,10 @@ public boolean configure(String name, Map<String, Object> params) throws Configu
             if (allocationAlgorithm != null) {
                 this.allocationAlgorithm = allocationAlgorithm;
             }
+            String strictness = configs.get("disk.provisioning.type.strictness");
```

Comment:
"disk.provisioning.type.strictness" is defined as dynamic and zone level configuration parameter. Getting the value from 'configs' and using a variable in configure() method will not make it dynamic or read from specific zone. Please use ConfigKey object with valuein() method to get the value at the exact usage point.

Reply:
Agree with @harikrishna-patnala , please use Config framework to get the zone-wide setting.

Reply:
I have moved the config to StorageManager